include/eelbot_framework/discord_bot/bot.hpp

```diff
@@ -39,14 +50,47 @@ class bot {
 	std::optional<std::string> ca_directory;
 	std::optional<std::string> http_proxy;
 
-	bool        gateway_active = false;
-	std::string session_id;
+	std::shared_ptr<websocket_client> ws_client;
+	std::shared_ptr<zlib_inflate>     z_inflator;
+
+	mutable std::mutex              gateway_mutex;
```

Comment:
this indentation, did a linter/IDE do it or did you manually align whitespace?

Reply:
Auto formatted with the rules defined in the `.clang-format` file.

Reply:
Is this particular rule (re: aligning this stuff) based off a popular guideline? It looks untidy TBH especially since the alignments for ht block above (ex: lines 53, 54 aren't the same)

Reply:
I'm not sure about its popularity. To have the same indentation with this particular rule, I would need to get rid of the line break in between.